src/npc_attack.cpp

```diff
@@ -486,23 +600,28 @@ npc_attack_rating npc_attack_throw::evaluate_tripoint(
     if( critter ) {
         att = source.attitude_to( *critter );
     }
-    const bool friendly_fire = att == Creature::Attitude::FRIENDLY &&
-                               !source.rules.has_flag( ally_rule::avoid_friendly_fire );
-    int attitude_mult = npc_attack_constants::attitude_multiplier.at( att );
-    if( friendly_fire ) {
-        // hitting a neutral creature isn't exactly desired, but it's a lot less than a friendly.
-        // if friendly fire is on, we don't care too much, though if an available hit doesn't damage them it would be better.
-        attitude_mult = npc_attack_constants::attitude_multiplier.at( Creature::Attitude::NEUTRAL );
+
+    if( att != Creature::Attitude::HOSTILE ) {
+        // No point in throwing stuff at neutral/friendly targets
+        return npc_attack_rating( cata::nullopt, location );
+    }
+
+    if( source.rules.has_flag( ally_rule::avoid_friendly_fire ) &&
+        !source.wont_hit_friend( location, thrown_item, false ) ) {
+        // Avoid friendy fire
+        return npc_attack_rating( cata::nullopt, location );
     }
 
     const float throw_mult = throw_cost( source, single_item ) * source.speed_rating() / 100.0f;
     const int damage = source.thrown_item_total_damage_raw( single_item );
     float dps = damage / throw_mult;
     const int distance_to_me = rl_dist( location, source.pos() );
-    const int distance_penalty = std::max( distance_to_me - 1,
-                                           ( source.closest_enemy_to_friendly_distance() - 1 ) * 2 );
+    const int distance_penalty = std::max( std::min( distance_to_me,
+                                           source.closest_enemy_to_friendly_distance() ) - 3, 0 );
+    const float suitable_item_mult = thrown_item.has_flag( flag_NPC_THROWN ) ? 0.2f : -0.15f;
+    const float distance_mult = -distance_penalty / 10.0f;
 
-    double potential = dps * attitude_mult - distance_penalty;
+    double potential = dps * ( 1.0f + distance_mult + suitable_item_mult );
     if( critter && damage >= critter->get_hp() ) {
```

Comment:
```suggestion
    if( potential > 0.0f && critter && damage >= critter->get_hp() ) {
```
And the same down below for that multiplication of potential.

Reply:
Done manually. I'll mark it as resolved.